pkg/controller/aws/s3/claim.go

```diff
@@ -42,6 +42,25 @@ var s3ACL = map[storagev1alpha1.PredefinedACL]s3.BucketCannedACL{
 	storagev1alpha1.ACLAuthenticatedRead: s3.BucketCannedACLAuthenticatedRead,
 }
 
+// AddDefault adds a controller that reconciles... TODO
+func AddDefault(mgr manager.Manager) error {
+	r := resource.NewDefaultReconciler(mgr,
```

Comment:
with this design, each concrete resource controller will create a controller to watch instances of the claim that they care about.  That feels a bit odd to me, to require each controller to duplicate this work.

Would it be possible to get this default class reconciler to run only ONCE per claim type? I'm not seeing anything in the default reconciler that is specific to the concrete resource, it looks like this could be done at the claim scope instead. Help me understand if that is not the case. thank you!

Reply:
This is a great point! My reason for implementing it the way it is currently preposed is simply a product of the structure of existing controllers. I agree that these controllers only need to be implemented once per claim type. @negz would love to hear if you see any drawbacks here, but I do not see any right off. Thanks for the feedback Jared!

Reply:
Ah, good point that did not occur to me. I agree one instance per claim type is the direction we should go.

How would you two feel about keeping the existing package structure for resource claim and managed resource controllers, but adding a new package, `default` under `pkg/controller/default` in which we'd declare the various add functions as `default.AddRedisCluster`, `default.AddPostgreSQLInstance`, `default.AddKubernetesCluster`, etc?

Reply:
Sounds good to me! That is actually how I have already begun to design it :)